release.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+vercomp () {
+	if [[ $1 == $2 ]]
+	then
+		return 0
+	fi
+	local IFS=.
+	local i ver1=($1) ver2=($2)
+	# fill empty fields in ver1 with zeros
+	for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+	do
+		ver1[i]=0
+	done
+	for ((i=0; i<${#ver1[@]}; i++))
+	do
+		if [[ -z ${ver2[i]} ]]
+		then
+			# fill empty fields in ver2 with zeros
+			ver2[i]=0
+		fi
+		if ((10#${ver1[i]} > 10#${ver2[i]}))
+		then
+			return 1
+		fi
+		if ((10#${ver1[i]} < 10#${ver2[i]}))
+		then
+			return 2
+		fi
+	done
+	return 0
+}
+
+# check if `gh` tool is installed.
+if ! command -v gh &> /dev/null
+then
+	echo "Github CLI tool is required, but could not be found."
+	echo "Install it via: $ brew install gh"
+	exit 1
+fi
+
+# check that we're on the `main` branch
+branch=$(git rev-parse --abbrev-ref HEAD)
+if [ branch != 'main' ] 
+then
+	echo "The 'main' branch must be the current branch out to make a release."
+	echo "You are currently on: $branch"
+	exit 1
+fi
+
+versionFile="./sources/Segment/version.swift"
```

Comment:
not sure if `V` needs to be uppercase

Reply:
case insensitive file system; but yeah it should be.